src/transformers/models/whisper/generation_whisper.py

```diff
@@ -773,8 +775,12 @@ def generate_with_fallback(
             generation_config.do_sample = temperature is not None and temperature > 0.0
 
             generation_config.temperature = temperature if generation_config.do_sample else 1.0
-            generation_config.num_beams = kwargs.pop("num_beams", 1) if not generation_config.do_sample else 1
+            generation_config.num_beams = kwargs.get("num_beams", 1) if not generation_config.do_sample else 1
 
+            generate_kwargs = dict(kwargs)
+            for key in ["do_sample", "temperature", "num_beams"]:
```

Comment:
`temperature` shouldn't be in `kwargs` as it's already an argument of `.generate` [here](https://github.com/huggingface/transformers/blob/923733c22bf4d3cc6661c8cd3b730b275e9a938e/src/transformers/models/whisper/generation_whisper.py#L269) right ?

It seems okay to check for `do_sample` and `num_beams` here

Reply:
Just wanted to be extra cautious here and make sure everything is safe locally, rather than relying on what gets passed down from 2 call frames up the stack. But I can remove `temperature` if you prefer.

Reply:
Looks good to me as is - there's a preference for more explicit handling of kwargs than more buried ones